pkg/daemon/ceph/osd/encryption.go

```diff
@@ -160,6 +160,17 @@ func dumpLUKS(context *clusterd.Context, disk string) (string, error) {
 	return cryptsetupOut, nil
 }
 
+func openLUKS(context *clusterd.Context, disk, target, passphrase string) error {
+	args := []string{"luksOpen", "--verbose", "--allow-discards", disk, target}
+	err := context.Executor.ExecuteCommandWithStdin(cryptsetupBinary, &passphrase, args...)
+	if err != nil {
+		logger.Errorf("failed to open LUKS device %q", disk)
+		return err
+	}
+
+	return nil
+}
```

Comment:
I notice this doesn't have a timeout. For the prepare job, I know we have been more lax on resource requests/timeouts in the past. IMO, it would still be good to have a timeout just to make sure we can continue with provisioning if a disk is slow to respond or some other reason. For LUKS, I think 2 1/2 minutes (150 seconds) should be good given that a device should normally open within 30 seconds.

@travisn wdyt?

Reply:
Why go 5 times the 30s? Either way the timeout seems plenty long so I don't have a strong opinion other than it seems like 30s would be sufficient.

Reply:
> Why go 5 times the 30s? Either way the timeout seems plenty long so I don't have a strong opinion other than it seems like 30s would be sufficient.

Set it to 30, I think that's sufficient.

Reply:
I suggest a higher value so that we don't get flakiness in our tests because of the value. A lot of times, when there are resource-constrainted environments, what is "supposed" to happen timing-wise, is much longer. I use 5x as a rule of thumb because when I've used 3x, there is often still flakiness. Maybe 120 seconds is good, but I would really suggest using more than 30.

Reply:
I did some digging. Luks uses a static 2 seconds to open a disk with a key, which is strange that it's static, but it apparently is so. There can be 8 key slots, which means it could take 16 total seconds for each unlock attempt with each key. Plus any overhead for CPU time. I believe the normal case is that the unlock should take less than 3 seconds, but let's not risk create a flaky system for our CI or for corner-case users by using too aggressive of a timeout. I still think having at least a 90 second timeout is reasonable.

The most common case will be that it finishes successfully in ~2 seconds. In the event of some corruption, it will take 1.5 minutes. If I/O to the disk is stalling, it will take 1.5 minutes for each disk to fail. For our CI, I/O stalling seems like it's not uncommon, and there are many users who try to run on R-Pis and other small systems.

On a different angle, if a server with 60 disks attached to a SAS controller has a controller failure such that I/Os stall indefinitely without a noticeable failure, the prepare job won't fail until after 1.5 hours. That is quite a while, but IMO if a user has that many disks on a node, they likely expect installation and upgrades will take a fair amount of time, and 1.5 hours isn't a lot compared to a rolling upgrade that could take a week for large clusters.

Reply:
changed it to 90,